plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -145,6 +145,97 @@ func NoDiskConflict(pod *api.Pod, existingPods []*api.Pod, node string) (bool, e
 	return true, nil
 }
 
+type MaxEBSVolumeCountChecker struct {
+	maxVolumes uint
+	pvInfo     PersistentVolumeInfo
+	pvcInfo    PersistentVolumeClaimInfo
+}
+
+// MaxEBSVolumeCountPredicate evaluates whether a pod can fit based on the number of AWS ElasticBlockStorage
+// volumes it requests, and those that are already present.  The maximum number is configurable, although
+// Amazon reccomends that only 40 volumes be mounted, including the system root volume.
+//
+// The predicate looks for both EBS volumes used directly, as well as PVC volumes that are backed by EBS volumes,
+// counts the number of unique volumes, and rejects the new pod if it would place the total count over the maximum.
+func NewMaxEBSVolumeCountPredicate(maxVolumes uint, pvInfo PersistentVolumeInfo, pvcInfo PersistentVolumeClaimInfo) algorithm.FitPredicate {
+	c := &MaxEBSVolumeCountChecker{
+		maxVolumes: maxVolumes,
+		pvInfo:     pvInfo,
+		pvcInfo:    pvcInfo,
+	}
+
+	return c.predicate
+}
+
+func (c *MaxEBSVolumeCountChecker) filterEBSVolumes(volumes []api.Volume, namespace string, ebsVolumes map[string]bool) error {
```

Comment:
You can transform this into a generic implementation by injecting a `func([]api.Volume, string) (map[string]bool, error)`, and then make the EBS checker inject this implementation.  There will likely be other people interested in using this predicate for their clouds.